src/runtime/pkg/oci/utils.go

```diff
@@ -919,6 +928,57 @@ func addHypervisorNetworkOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfi
 	})
 }
 
+func addHypervisorInitdataOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfig) error {
+	if value, ok := ocispec.Annotations[vcAnnotations.Initdata]; ok {
+		initdataTomlGzip, err := base64.StdEncoding.DecodeString(value)
+		if err != nil {
+			ociLog.Errorf("initdata uncompress failed! %v", err)
+			return err
+		}
+
+		gzipReader, err := gzip.NewReader(bytes.NewReader(initdataTomlGzip))
+		if err != nil {
+			return fmt.Errorf("create gzip reader failed: %v", err)
+		}
+		defer gzipReader.Close()
+
+		initdataToml, err := io.ReadAll(gzipReader)
+		if err != nil {
+			return fmt.Errorf("uncompressing initdata with gzip error: %v", err)
+		}
+
+		initdataStr := string(initdataToml)
+		var initdata kataTypes.Initdata
+		if _, err := toml.Decode(initdataStr, &initdata); err != nil {
+			return err
+		}
```

Comment:
It could be a bit more elegant to work with `io.Reader`s here - something like

```go
b64Reader := base64.NewDecoder(base64.StdEncoding, bytes.NewReader(value))
gzipReader, err := gzip.NewReader(b64Reader)
_, err = toml.NewDecoder(gzipReader).Decode(&initdata)
```

Reply:
Nice suggestion. Some dependency
```
gzip(b64(initdata)) ---gzipReader---> b64(initdata) ----b64Reader----> initdata(plaintext) ----toml----> initdata
                                                                                                                                          |                                            |
                                                                                                                                     calculate hash                          |
                                                                                                                                          +-------------------------+
```

So we might need to drop the last toml reader and cache the initdata plaintext in memory

Reply:
Ah, true - but at least we can skip the intermediate gzip bytes.